src/libpriv/rpmostree-core.c

```diff
@@ -945,7 +947,7 @@ rpmostree_context_download_metadata (RpmOstreeContext *self,
       if (!dnf_repo_check(repo,
                           dnf_context_get_cache_age (self->hifctx),
                           hifstate,
-                          NULL))
+                          NULL) || !cache_suffix_match)
```

Comment:
`dnf_repo_check` should just work as before, right? If the location is pointing to a new unpopulated cachedir, then it should return `FALSE` so we can update it.